src/definition/guessfunc.go

```diff
@@ -0,0 +1,15 @@
+package definition
+
+import (
+	"fmt"
+	"strings"
+)
+
+// K8sMetricSetTypeGuesser is the metric set type guesser for k8s integrations.
```

Comment:
since the behavior is not straightforward can we add a few more lines?

Reply:
I simply moved this function (it already existed) to avoid a circular dependency, but you are totally right, it is quite confusing and I will take the chance to extend the comment.

Reply:
Updated the comment here: [65c1156](https://github.com/newrelic/nri-kubernetes/pull/577/commits/65c11564ad5aae1b114fdac74f44ba23e493ca88)